edx_sga/static/js/src/edx_sga.js

```diff
@@ -314,10 +324,80 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                     });
                 block.find('#staff-debug-info-button')
                     .leanModal();
+
+                $(element).find('#download-init-button').click(function(e) {
+                  e.preventDefault();
+                  var self = this;
+                  $.get(
+                    prepareDownloadSubmissionsUrl,
+                    function(data) {
+                      if (data["downloadable"]) {
+                        window.location = downloadSubmissionsUrl;
+                        $(self).removeClass("disabled");
+                      } else {
+                        $(self).addClass("disabled");
+                        $(element).find('.task-message')
+                          .show()
+                          .html(preparingSubmissionsMsg)
+                          .removeClass("ready-msg")
+                          .addClass("preparing-msg");
+                          pollSubmissionDownload(downloadSubmissionsStatusUrl, preparingSubmissionsMsg);
+                      }
+                    }
+                  );
+                });
             }
         });
     }
 
+    function pollSubmissionDownload(downloadSubmissionsStatusUrl, preparingSubmissionsMsg) {
+      pollUntilSuccess(downloadSubmissionsStatusUrl, checkResponse, 10000, 400).then(function(res) {
```

Comment:
1. Sorry, I should have noticed this before. There's a misunderstanding here about `pollUntilSuccess`. `pollUntilSuccess` will resolve the promise if `checkResponse` function returns true before the number of maximum tries is exceeded or there is an error response. At this point in the execution we already know if the zip file was successfully found. `.then()` is for our success handler function, and `.fail()` is our failure handler function. Failure means that we either exceeded the maximum tries or there was an error response from the server. In our success function, it doesn't make any sense to check for `if (res["zip_available"])` - the fact that the function is executing means that the promise resolved, which means that the polling function found the zip file. The `.then()` handler function should just be lines 356-361. 
1. The error message at line 370 should be passed into `gettext()`
1. It's not necessary to pass in `downloadSubmissionsStatusUrl` and `preparingSubmissionsMsg` to this function, and I don't think it makes much sense. I think it's fine for this function to get these values from the outer context
1. I think we should also drop the maximum retries from 400 to 100. That will poll the endpoint for a little over 16 minutes. I think that's plenty of time.